pkg/apis/networking/types.go

```diff
@@ -566,31 +566,24 @@ type ClusterCIDRConfigSpec struct {
 	// +optional
 	NodeSelector *api.NodeSelector
 
-	// IPv4 defines the IPv4 CIDR and the PerNodeMaskSize.
-	// At least one of IPv4 or IPv6 must be provided. If both are
-	// provided, the number of IPs allocated to each must be the same
-	// (32 - ipv4.perNodeMaskSize).
+	// PerNodeHostBits defines the host bits to be configured per node.
+	// A subnet mask determines how much of the address is used for network bits
+	// and host bits. For example and IPv4 address of 192.168.0.0/24, splits the
+	// address into 24 bits for the network portion and 8 bits for the host portion.
+	// For a /24 mask for IPv4 or a /120 for IPv6, configure PerNodeHostBits=8
 	// This field is immutable.
 	// +optional
-	IPv4 *CIDRConfig
+	PerNodeHostBits int32
 
-	// IPv6 defines the IPv4 CIDR and the PerNodeMaskSize.
-	// At least one of IPv4 or IPv6 must be provided. If both are
-	// provided, the number of IPs allocated to each must be the same
-	// (128 - ipv6.perNodeMaskSize).
+	// IPv4CIDR defines an IPv4 IP block in CIDR notation(e.g. "10.0.0.0/8").
 	// This field is immutable.
 	// +optional
-	IPv6 *CIDRConfig
-}
-
-// CIDRConfig defines the CIDR and Mask size per IP Family(IPv4/IPv6).
-type CIDRConfig struct {
-	// An IP block in CIDR notation ("10.0.0.0/8", "fd12:3456:789a:1::/64").
-	CIDR string
+	IPv4CIDR string
```

Comment:
Do we enforce that v4 and v6 CIDRs allocate the same number of nodes?  Not super important today but we should decide whether that is something we will validate?

e.g.

is this allowed?

```
perNodeHostBits: 8
ipv4CIDR: 10.0.0.0/16
ipv6CIDR: 10::/64
```

Reply:
Yes currently there is no validation for same number of nodes, so the example above is allowed.

I think yes we should decide on the validation, as it might lead to wastage of IPv6 CIDRs(unless we decide to implement a range allocator which will continue assigning IPv6 only CIDRs when IPv4 are exhausted, this is not the behavior currently)

Reply:
It's totally allowed, I think by design. The assumption was that you might have a fragmented IPv4 space but a pretty large IPv6 space and it doesn't make sense to have to cut up the IPv6 space manually

There's no wastage as the same CIDR can show up in multiple CCC objects.